### PR TITLE
Adds unit testing with vitest and react testing library

### PR DESCRIPTION
**Pull request by windyScripts**
Adds vitest and react testing library as devdependencies, a script called test to run vitest and a clause in the ci github actions to run tests

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -34,4 +34,6 @@ jobs:
           echo NEXT_PUBLIC_CONVEX_URL=${{ secrets.NEXT_PUBLIC_CONVEX_URL }} >> .env.local
           npm run build
       # - name: Run test with coverage
-      #   run: pnpm run coverage
\ No newline at end of file
+      #   run: pnpm run coverage
```
Comment (alcpereira) [inline .github/workflows/ci.yaml:37]: We're not doing the coverage yet, but I think we can implement it in a separate PR.

**Review by alcpereira — APPROVED** [review]
Please just remove the `Closes` issue and edit the title as we've not finished 100% of the requirements (React Testing Library for example).

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (windyScripts) [reply .github/workflows/ci.yaml:37]: Alright, changes have been made.

**Pull request merged**